python/packages/autogen-ext/src/autogen_ext/task/_agent_name_termination.py

```diff
@@ -0,0 +1,38 @@
+from typing import Sequence, List
+
+from autogen_agentchat.base import TerminationCondition, TerminatedException
+from autogen_agentchat.messages import StopMessage, AgentMessage, ChatMessage
+
+
+class AgentNameTermination(TerminationCondition):
+    """Terminate the conversation after a specific agent responds.
+
+    Args:
+        agents (List[str]): List of agent names to terminate the conversation.
+
+    Raises:
+        TerminatedException: If the termination condition has already been reached.
+    """
+
+    def __init__(self, agents: List[str]) -> None:
+        self._agents = agents
+        self._terminated = False
+
+    @property
+    def terminated(self) -> bool:
+        return self._terminated
+
+    async def __call__(self, messages: Sequence[AgentMessage]) -> StopMessage | None:
+        if self._terminated:
+            raise TerminatedException("Termination condition has already been reached")
+        if not messages:
+            return None
+        last_message = messages[-1]
+        if last_message.source in self._agents:
+            if isinstance(last_message, ChatMessage):
```

Comment:
This check is not necessary, as messages from the same agent are consecutive.

Reply:
Thanks, but can the last message be tool call message or tool call result message? i add this check just to make sure tool call is done and agent recieved the tool call result. If it truly unecessary, i will remove it. please let me know.

Reply:
The tool calls messages are internal to the agent and must conclude before agent responds, after that the messages are processed by the termination. So, the termination condition is not currently applied within each agent's on_messages method, it gets a batch of messages once the response is produced, which always contans a `ChatMessage` at the end. 

Though it does require users to type-check their code though, as we don't validate the message type in group chat.